File: fibonacci.py
"""Fibonacci number utilities."""


def fib(n: int) -> int:
    """
    Return the n-th Fibonacci number (0-indexed: fib(0)==0, fib(1)==1).
    """
    if n < 0:
        raise ValueError("n must be non-negative")
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a


if __name__ == "__main__":
    import sys

    k = int(sys.argv[1]) if len(sys.argv) > 1 else 10
    print(fib(k))

File: fibonacci.js
/**
 * Returns the n-th Fibonacci number using the definition F(0) = 0, F(1) = 1.
 * @param {number} n - Non-negative integer index
 * @returns {bigint}
 */
function fibonacci(n) {
  if (!Number.isInteger(n) || n < 0) {
    throw new TypeError("n must be a non-negative integer");
  }
  if (n <= 1) {
    return BigInt(n);
  }
  let a = 0n;
  let b = 1n;
  for (let i = 2; i <= n; i += 1) {
    const next = a + b;
    a = b;
    b = next;
  }
  return b;
}

module.exports = { fibonacci };
